### PR TITLE
chore: hash device id

### PR DESCRIPTION
**Pull request by tmspzz**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Device id can be sensitive and we would like to be a little more resilient to accidents

### Solutions

Limit device id collection to device running API lvl 26 or more and hash it before associating it with DataDog

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/WireApplication.kt [hunk 1]
```diff
@@ -116,7 +117,11 @@ class WireApplication : Application(), Configuration.Provider {
 
         val credentials = Credentials(clientToken, environmentName, appVariantName, applicationId)
         Datadog.initialize(this, credentials, configuration, TrackingConsent.GRANTED)
-        Datadog.setUserInfo(id = getDeviceId(this))
+        
+        if (android.os.Build.VERSION.SDK_INT >= 26) {
```
Comment (ohassine) [inline app/src/main/kotlin/com/wire/android/WireApplication.kt]: We can use `Build.VERSION_CODES.O` instead a magic number

**Review by ohassine — APPROVED** [review]

File: app/src/main/kotlin/com/wire/android/WireApplication.kt [hunk 2]
```diff
@@ -116,7 +117,11 @@ class WireApplication : Application(), Configuration.Provider {
...
+            Datadog.setUserInfo(id = getDeviceId(this)?.sha256())
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/WireApplication.kt]: we need to change getDeviceId to read the device id only for 26+
and the app is still reading the device id in other places  like `WireApplication`

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
the app is still reading the device id in other places like WireApplication

**Review by MohamadJaara — APPROVED** [review]
LGTM

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-973/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-973/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-973/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-973/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [13](https://10.10.124.134/job/wire-android-reloaded/job/PR-973/13/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4419229)(4419229)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4419229)(4419229)-dev-release.apk)

**Pull request merged**